go/vt/mysqlctl/mysqld.go

```diff
@@ -676,6 +679,32 @@ func (mysqld *Mysqld) Init(ctx context.Context, cnf *Mycnf, initDBSQLFile string
 	return nil
 }
 
+// For debugging purposes show the last few lines of the MySQL error log.
+// Return a suggestion (string) if the file is non regular or can not be opened.
+// This helps prevent cases where the error log is symlinked to /dev/stderr etc,
+// In which case the user can manually open the file.
+func readTailOfMysqldErrorLog(fileName string) string {
+	file, err := os.Open(fileName)
```

Comment:
Move the `os.Open` call down to after the check for a regular file. There's no need to open it until we have decided we're going to try to read from it.

Also, after passing the error check for `Open`, immediately `defer file.Close()` so any return path cleans up.

Reply:
Done, with one exception the `staticcheck` asked for the `defer file.Close()` to be after the error check, rather than immediately after the open.